antiplag/models.py

```diff
@@ -17,6 +19,24 @@ class SubmissionStatus(models.TextChoices):
     def __str__(self):
         return f"Submission {self.id} ({self.status})"
 
+    @classmethod
+    def count(cls):
+        cached_value = cache.get("submission_count", None)
+        return cached_value if cached_value else cls.objects.count()
+
+    @classmethod
+    def average_time(cls):
+        cached_value = cache.get("submission_avg_time")
+        if cached_value is not None:
+            return cached_value
+        else:
+            result = cls.objects.filter(
+                status=cls.SubmissionStatus.PROCESSED
+            ).aggregate(
+                average_time=models.Avg(models.F("updated_at") - models.F("created_at"))
+            )
+            return ceil(result.get("average_time").seconds / 60)  # in minutes
+
```

Comment:
Am I missing something? Don't you need to set the cache as well?

Reply:
![image](https://user-images.githubusercontent.com/27869979/113829428-c6ff5980-9785-11eb-9921-5adbf17e768a.png)